created-profile-model

GurkirtSingh opened the pull request:
### What this PR does (required):
- A Profile model is created to store user's details, location and availability.
- Link to ticket https://hatchways.notion.site/Profile-Model-df539b4588b14eb0b21fb4c878b6ebe6

GurkirtSingh committed:
created-profile-model

A Profile model is created to store user's detail, location and availability.

ugur-akin commented on server/models/Profile.js:
```diff
@@ -0,0 +1,61 @@
+const mongoose = require("mongoose");
+
+const profileSchema = new mongoose.Schema({
+  user_id: {
```
We generally prefer `camelCase` for field names in JS: for example `user_id` would become `userId` and `firstname` would be `firstName`. Let's fix all the properties accordingly 👍🏼

GurkirtSingh replied:
Alright I'll follow camelCase pattern for fields from now on.

ugur-akin commented on server/models/Profile.js:
```diff
@@ -0,0 +1,61 @@
+
+const profileSchema = new mongoose.Schema({
+  user_id: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
```
There shuould be a `ref` SchemaType here as well ([relevant mongoose docs](https://mongoosejs.com/docs/populate.html#populate)) 👍🏼

GurkirtSingh replied:
Agree with you. I'll fix it

ugur-akin commented on server/models/Profile.js:
```diff
@@ -0,0 +1,61 @@
+      type: String,
+      required: true,
+    },
+    index: true,
```
I like the idea🙌🏼 Although I'm personally not 💯 if `index` works in nested paths, would be a nice exercise to find out 🤔

ugur-akin commented on server/models/Profile.js:
```diff
@@ -0,0 +1,61 @@
+      },
+    },
+  ],
+  hourlyWage: {
```
This is subtle but I don't think `wage` is the best choice of word for this. Maybe `hourlyRate` or `hourlyFee` instead? What do you think?

GurkirtSingh replied:
Yeah that was confusing name 😃. Will fix it.

ugur-akin reviewed (commented):
Good start 🚀 The model looks good overall, there are just a few minor issues to address.

I noticed we didn't actualy compile and/or export a model from the schema defined here. I suggest taking a look at the [docs](https://mongoosejs.com/docs/models.html#compiling) and add the code for those as well 👍🏼 

As a bonus, it would have been nice to some validation as well. Just to give some ideas, we could set some boundries on the birth date, we could check `end > start` for availability blocks or we could use regex validation for url/phone number.

GurkirtSingh commented:
Thank for the review @ugur-akin. Yes, it was mistake to not export the model I will do it in next commit. Also I'll add validation for the fields. 👨🏽‍🔧

GurkirtSingh committed:
Added validation on Profile Schema

Some of Profile Model fields need validation such as  birthdate, phone number and availability dates

Also replaced field name with camelCase

finally exported Profile Model

moffatethan commented on server/models/Profile.js:
```diff
@@ -0,0 +1,91 @@
+  return value > this.startAt;
+};
+
+profileSchema
+  .path("phoneNumber")
+  .validate(phoneNumberValidator, "`{VALUE}` is not valid phone number");
+profileSchema
+  .path("photoUrl")
+  .validate(photoUrlValidator, "`{VALUE}` is not valid photo url");
```
💯

moffatethan reviewed (approved):
LGTM

GurkirtSingh commented:
Perfect, then 👨🏽‍💻 I'll merge it.
